<launch>

  <!-- launch qr tracking for camera pose estimation -->
  <arg name="marker_size" default="4.4" />
  <arg name="max_new_marker_error" default="0.08" />
  <arg name="max_track_error" default="0.2" />
  
  <arg name="cam_image_topic" default="/camera/depth_registered/points" />
  <arg name="cam_info_topic" default="/camera/rgb/camera_info" />		
  <arg name="output_frame" default="/camera_rgb_frame" />
  
  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkers" respawn="false" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame)" />

  <!-- Perception Server -->
  <node name="perception" pkg="perception" type="perception" output="screen"/>

  <!-- Moveit Visualizations -->
  <param name="robot_description_semantic" textfile="$(find baxter_moveit_config)/config/baxter.srdf"/>
  <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 /world /base 100" />

  <!-- launch rviz -->
  <node name="$(anon rviz)" pkg="rviz" type="rviz" respawn="true" output="screen"
	args="-d $(find perception)/launch/interface.rviz">
  </node>
</launch>
